core/services/feeds/service.go

```diff
@@ -178,6 +192,56 @@ func (s *service) GetJobProposal(id int64) (*JobProposal, error) {
 	return s.orm.GetJobProposal(context.Background(), id)
 }
 
+func (s *service) ApproveJobProposal(ctx context.Context, id int64) error {
+	if s.fmsClient == nil {
+		return errors.New("fms rpc client is not connected")
+	}
+
+	jp, err := s.orm.GetJobProposal(ctx, id)
+	if err != nil {
+		return err
+	}
+
+	if jp.Status != JobProposalStatusPending {
+		return errors.New("must be a pending job proposal")
+	}
+
+	ctx, cancel := context.WithTimeout(ctx, postgres.DefaultQueryTimeout)
+	defer cancel()
+
+	j, err := s.generateJob(jp.Spec)
+	if err != nil {
+		return err
```

Comment:
might be worth wrapping some/all of those errors being returned

Reply:
Added wrapping to the errors